### PR TITLE
GH-311 - Fields without a label definition in the dictionary should not be shown in GUI

### PR DESCRIPTION
**Pull request by evdhoudt**
At the moment only for Organizations the labels are named in the dictionary as far as I could see.

### CODE REVIEW

File: anet.yml [hunk 1]
```diff
@@ -155,15 +155,16 @@ dictionary:
...
-  PRINCIPAL_ORG_LABEL_LONGNAME: Official Organization Name
   PRINCIPAL_ORG_PLACEHOLDER_LONGNAME: e.g. Afghan Ministry of Defense
-  ADVISOR_ORG_LABEL_LONGNAME: Description
   ADVISOR_ORG_PLACEHOLDER_LONGNAME: e.g. Force Sustainment
-  PRINCIPAL_ORG_LABEL_IDENTIFICATIONCODE: Identification Code
   PRINCIPAL_ORG_PLACEHOLDER_IDENTIFICATIONCODE: optional
-  ADVISOR_ORG_LABEL_IDENTIFICATIONCODE: UIC
   ADVISOR_ORG_PLACEHOLDER_IDENTIFICATIONCODE: the six character code
   SUPPORT_EMAIL_ADDR: support@example.com
+  field_labels:
```
Comment (gjvoosten) [inline anet.yml]: The labels and placeholders are now too far apart; maybe have something like:
```
  fields:
    labels:
      ADVISOR_ORG_LONGNAME: fred
      …
    placeholders:
      ADVISOR_ORG_LONGNAME: ask fred
      …
```
?

Comment (VassilIordanov) [reply anet.yml]: or maybe even:
```
  fields:
      ADVISOR_ORG_LONGNAME: 
         label: fred
         placeholder: ask fred
      …
```

Comment (evdhoudt) [reply anet.yml]: @gjvoosten shall I go with @VassilIordanov 's format?

Comment (gjvoosten) [reply anet.yml]: I have a slight dislike of the alternative (that I also considered, but rejected), as it is too repetetive (each key has subkeys `label` and `placeholder`).

Comment (VassilIordanov) [reply anet.yml]: doesn't matter to me that much. I would go with the former if these are the only two parameters we are planning to use. I would go with the latter if we plan to evolve it into something like:
```
  fields:
      TaskCustom: 
         label: Task status
         placeholder: ask fred
         Enum:
            OPEN: Open
            CLOSED: Closed
         Hint: Task status is blah blah blah
```

Comment (VassilIordanov) [reply anet.yml]: I think if we keep all data pertaining to the field together, it is indeed more verbose, but there is less room for mistakes. In the current structure, we already have fields with different requirements, such as:
```  fields:
      ADVISOR_ORG_LONGNAME: 
         label: fred
         placeholder: ask fred
      TASK_CUSTOM_FIELD:
         label: xyz
         placeholder: xyz
      TASK_CUSTOM_ENUM: 
         label: Task status
         Enum:
             OPEN: Open
             CLOSED: Closed
    ....```

Comment (gjvoosten) [reply anet.yml]: Then let's go all the way.

File: anet.yml [hunk 2]
```diff
@@ -150,21 +150,23 @@ dictionary:
...
-  PRINCIPAL_ORG_PLACEHOLDER_IDENTIFICATIONCODE: optional
-  ADVISOR_ORG_PLACEHOLDER_IDENTIFICATIONCODE: the six character code
   SUPPORT_EMAIL_ADDR: support@example.com
-  field_labels:
-    ADVISOR_ORG_LABEL_LONGNAME: Description
-    PRINCIPAL_ORG_LABEL_IDENTIFICATIONCODE: Identification Code
-    PRINCIPAL_ORG_LABEL_LONGNAME: Official Organization Name
-    ADVISOR_ORG_LABEL_IDENTIFICATIONCODE: UIC
+  fields:
+    ADVISOR_ORG:
```
Comment (VassilIordanov) [inline anet.yml]: ADVISOR_ORG is not a field.
Under fields we should have a flat list of fields, each one of them with a label and a placeholder

Comment (VassilIordanov) [reply anet.yml]: I'll give it a try and send an example....

**Comment by VassilIordanov** [comment]
@evdhoudt @gjvoosten @maradragan Getting there. Could I have an intermediate review?

What I am interested in is - is the "dictionary" moving in the right direction.

still to do:
- make a schema for dictionary (and possibly externalize it as a json file)
- update documentation (possibly based on the schema)
- longer term: decide if we should majically inject the dictionary as props in the Components (as DictionaryField starts to do), or keep explicitly referencing them

File: anet.yml [hunk 3]
```diff
@@ -151,112 +151,136 @@ logging:
...
   SUPPORT_EMAIL_ADDR: support@example.com
+  fields:
+    task:
+      shortLabel: Task
+      longLabel: Tasks and Milestones
+      projectedCompletion:
+        label: Projected Completion
+        placeholder: 'Fill in the projected completion date'
+      plannedCompletion:
+        label: 'Planned comletion'
```
Comment (gjvoosten) [inline anet.yml]: Fix typo: Planned com*p*letion.

File: client/src/pages/reports/Show.js [hunk 4]
```diff
@@ -217,10 +217,10 @@ export default class ReportShow extends Page {
 						<Form.Field id="author" label="Report author">
 							<LinkTo person={report.author} />
 						</Form.Field>
-						<Form.Field id="advisorOrg" label={dict.lookup('ADVISOR_ORG_NAME')}>
+						<Form.Field id="advisorOrg" label={Settings.fields.ADVISOR_ORG.name}>
```
Comment (gjvoosten) [inline client/src/pages/reports/Show.js]: Change to:
```
                                                <Form.Field id="advisorOrg" label={Settings.fields.advisor.org.name}>
```

File: src/main/java/mil/dds/anet/AnetApplication.java [hunk 5]
```diff
@@ -114,7 +120,12 @@ public void run(AnetConfiguration configuration, Environment environment) {
 		final DBIFactory factory = new DBIFactory();
 		final DBI jdbi = factory.build(environment, configuration.getDataSourceFactory(), "mssql");
 
-		logger.info("dictionary: {}", new JSONObject(configuration.getDictionary()).toString(2));
+		final JSONObject dictionary = getDictionary(configuration);
+		if (dictionary == null) {
+			// Can't continue without a dictionary
+			System.exit(1);
```
Comment (VassilIordanov) [inline src/main/java/mil/dds/anet/AnetApplication.java]: provide feedback as off why can't continue

File: src/main/java/mil/dds/anet/AnetApplication.java [hunk 6]
```diff
@@ -114,7 +120,12 @@ public void run(AnetConfiguration configuration, Environment environment) {
 		final DBIFactory factory = new DBIFactory();
 		final DBI jdbi = factory.build(environment, configuration.getDataSourceFactory(), "mssql");
 
-		logger.info("dictionary: {}", new JSONObject(configuration.getDictionary()).toString(2));
+		final JSONObject dictionary = getDictionary(configuration);
```
Comment (VassilIordanov) [inline src/main/java/mil/dds/anet/AnetApplication.java]: This makes the dictionary structure being checked when doing `anet start anet.yml` . It should also happen during `anet check anet.yml` - possilby by changing/replacing the `CheckCommand.java` behavior

**Comment by evdhoudt** [comment]
@VassilIordanov  and @gjvoosten I'm not certain what still needs to be done to get this issue closed? Do I need to pick anything up?

**Comment by VassilIordanov** [comment]
@evdhoudt I think it is ok for now, I am finishing up the dictionary refactoring and will then ask for final review before merging.

**Comment by VassilIordanov** [comment]
I think it is best to merge this as is. I will continue work in another branch/issue to expand the dictionary to cover more values that are currently hard coded.
Could you please review? @gjvoosten @maradragan

**Pull request merged**